### PR TITLE
docs: add arazzo rules documentation

### PR DESCRIPTION
**Pull request by DmitryAnansky**
## What/Why/How?

Add documentation for Arazzo and AsyncAPI rules

## Reference

Closes: https://github.com/Redocly/redocly-cli/issues/1691

## Check yourself

- [ ] Code changed? - Tested with redoc/reference-docs/workflows (internal)
- [ ] All new/updated code is covered with tests
- [ ] New package installed? - Tested in different environments (browser/node)

## Security

- [x] Security impact of change has been considered
- [x] Code follows company security practices and guidelines

### CODE REVIEW

File: docs/rules/recommended.md [hunk 1]
```diff
@@ -27,6 +27,12 @@ Errors:
 - [security-defined](./security-defined.md)
 - [spec-components-invalid-map-name](./spec-components-invalid-map-name.md)
 - [spec](./spec.md)
+- [parameters-unique](./arazzo/parameters-unique.md)
```
Comment (DmitryAnansky) [inline docs/rules/recommended.md:30]: It seems to me that we need different structure for rules documentation.
Current version does not diverse OAS rules from AsyncAPI and Arazzo rules.
Its easy to be confused and mix spec specific rules, without checking rule details.

cc: @lornajane , @tatomyr

File: docs/rules/spec.md [hunk 2]
```diff
@@ -12,6 +12,15 @@ Ensures that your API document conforms to the [OpenAPI specification](https://s
 | 3.0 | ✅            |
 | 3.1 | ✅            |
 
+| ARAZZO | Compatibility |
```
Comment (DmitryAnansky) [inline docs/rules/spec.md:15]: I added Arazzo and AsyncAPI section to start discussion on refactoring the structure.
This example with `spec.md` displays the biggest confusion as all supported standards have `spec` rule, but this page is reserved for OAS.

cc: @lornajane , @tatomyr

File: packages/core/src/rules/arazzo/index.ts [hunk 3]
```diff
@@ -19,7 +19,7 @@ export const rules: ArazzoRuleSet<'built-in'> = {
   spec: Spec as ArazzoRule,
   assertions: Assertions as ArazzoRule,
   'parameters-not-in-body': ParametersNotInBody as ArazzoRule,
-  'sourceDescription-type': SourceDescriptionType as ArazzoRule,
+  'sourceDescription-type': SourceDescriptionsType as ArazzoRule,
```
Comment (DmitryAnansky) [inline packages/core/src/rules/arazzo/index.ts:22]: Just rename file to have one name pattern.

File: docs/guides/lint-asyncapi.md [hunk 4]
```diff
@@ -53,8 +53,8 @@ some of the built-in rules. The currently-supported rules are:
 
 - `info-contact`: the `Info` section must contain a valid `Contact` field.
 - `operation-operationId`: every operation must have a valid `operationId`.
-- `channels-kebab-case`: channel names should be `kebab-case` (lowercase with hyphens).
-- `no-channel-trailing-slash`: channel names must not have trailing slashes in their names.
+- `channels-kebab-case`: channel address should be `kebab-case` (lowercase with hyphens).
```
Comment (DmitryAnansky) [inline docs/guides/lint-asyncapi.md:56]: I was confused when I didn't find `name` property in channel, I guess it should be `address`.
Please correct me if I am wrong, @lornajane | @tatomyr .

**Comment by jeremyfiel** [comment]
sorry to give duplicate work but most of the arazzo defined rules in this pr are handled by the recently merged JSON Schema definition for Arazzo 1.0.   It's currently in the 1.0.0-dev branch but is expected to be merged to main very soon.

https://github.com/OAI/Arazzo-Specification/pull/224/files#diff-c3b79a68b0d389644801b367021ceb2169c0a1afbdec651107b11605db4671c2

File: docs/rules/arazzo/parameters-unique.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
+---
+slug: /docs/cli/rules/arazzo/parameters-unique
+---
+
+# parameters-unique
+
+Requires the `parameters` list not to include duplicate parameters.
+
+| ARAZZO | Compatibility |
```
Comment (adamaltman) [inline docs/rules/arazzo/parameters-unique.md:9]:
```suggestion
| Arazzo | Compatibility |
```

File: docs/rules/arazzo/parameters-unique.md [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+workflows:
+  - workflowId: get-museum-hours
+    parameters:
+      - in: header
+        name: Authorization
+        value: Basic Og==
+      - in: header
+        name: Auth
+        value: Main Og==
+```
```
Comment (adamaltman) [inline docs/rules/arazzo/parameters-unique.md:89]: I'm trying to see the difference between these and I see one has a header "Auth" and the other one has two "Authorization" headers. What does this have to do with "licenses"?

File: docs/rules/recommended.md (hunk 1 above)
Comment (adamaltman) [reply docs/rules/recommended.md:30]: I agree. Maybe groups like "OpenAPI", and  "Arazzo".

File: docs/rules/spec.md (hunk 2 above)
Comment (adamaltman) [reply docs/rules/spec.md:15]:
```suggestion
| Arazzo | Compatibility |
```

File: docs/guides/lint-asyncapi.md (hunk 4 above)
Comment (adamaltman) [reply docs/guides/lint-asyncapi.md:56]: Seems like we could also have an "AsyncAPI" group?

File: docs/rules/minimal.md [hunk 7]
```diff
@@ -12,6 +12,8 @@ Errors:
 - [no-server-variables-empty-enum](./no-server-variables-empty-enum.md)
 - [no-unresolved-refs](./no-unresolved-refs.md)
 - [spec](./spec.md)
+- [stepId-unique](./arazzo/stepId-unique.md)
+- [workflowId-unique](./arazzo/workflowId-unique.md)
```
[the remark this answers is not in the document]
Comment (adamaltman) [reply docs/rules/minimal.md:16]: Yeah.

File: docs/rules/recommended.md (hunk 1 above)
Comment (adamaltman) [reply docs/rules/recommended.md:30]: Or separate labels.

File: docs/rules/arazzo/parameters-unique.md (hunk 6 above)
Comment (DmitryAnansky) [reply docs/rules/arazzo/parameters-unique.md:89]: I wanted to illustrate that parameters can't have the same name under one `parameters` section, maybe the correct example can be different.

**Comment by DmitryAnansky** [comment]
> sorry to give duplicate work but most of the arazzo defined rules in this pr are handled by the recently merged JSON Schema definition for Arazzo 1.0. It's currently in the 1.0.0-dev branch but is expected to be merged to main very soon.
> 
> https://github.com/OAI/Arazzo-Specification/pull/224/files#diff-c3b79a68b0d389644801b367021ceb2169c0a1afbdec651107b11605db4671c2

Thank you for your contribution, having an official JSON Schema is great and will be applied for generating types,
but we'll use these rules to validate internal type mapping, as the JSON schema might not fully address some nested cases or apply entirely to the internally mapped types.

File: docs/rules/arazzo/parameters-unique.md (hunk 6 above)
Comment (jeremyfiel) [reply docs/rules/arazzo/parameters-unique.md:89]: The schema should handle these particular constraints without additional rules.

Comment (tatomyr) [reply docs/rules/arazzo/parameters-unique.md:89]: @jeremyfiel could you explain how exactly the schema enforces unique parameter names?

Comment (jeremyfiel) [reply docs/rules/arazzo/parameters-unique.md:89]: hmm.. i suppose you may have a point here.. while it has `uniqueItems: true`, it doesn't mean you can't have one parameter with two different values. 

```yaml
parameters:
  - name: test
    in: header
    value: foo
  - name: test
    in: header
    value: bar
```

You're right!

this was proposed in JSON Schema as a new vocabulary but hasn't been included in the latest draft.  It's currently a custom keyword implementation.  https://github.com/json-schema-org/json-schema-vocabularies/issues/22

File: docs/rules/minimal.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (lornajane) [reply docs/rules/minimal.md:16]: Yes, we really, really need this. We also really need 3.0 and 3.1 especially as 3.2 is soon and will require some more (minor) alterations.

[the remark this answers is not in the document]
Comment (lornajane) [reply docs/rules/minimal.md:16]: We know what format of description we have, so we could create recommended/etc rulesets for each option and have them be the zero-config defaults.

File: docs/rules/recommended.md (hunk 1 above)
Comment (lornajane) [reply docs/rules/recommended.md:30]: Yes, definitely. I'll push some changes to this branch and try to make it easier to understand.

File: docs/rules/spec.md (hunk 2 above)
Comment (lornajane) [reply docs/rules/spec.md]: It is confusing, and quite a few of the rules work for more than one format (like info-contact I think for AsyncAPI)

File: docs/rules/spot/parameters-not-in-body.md [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+flowchart TD
+
+Root ==> workflows --> workflow --> steps --> step --> parameters
+
+style Contact fill:#codaf9,stroke:#0044d4,stroke-width:5px
+```
+
+## API design principles
+
+This is `Spot` specific rule.
```
Comment (lornajane) [inline docs/rules/spot/parameters-not-in-body.md:39]: We might want to label it as a separate spot version compatibility rather than Arazzo compatibility. Is the current version of spot a `v2` ?

**Review by lornajane — COMMENTED** [review]
Thanks for creating the pages for each rule, this is a great start! I will do some copy editing (I'll just push the changes if you don't mind), and figure out how to structure the information for the different description formats, but I need a few more things from you @DmitryAnansky  if possible:
- answer the open questions on the PR, I can transfer the answer to all the pages where it's something that comes up repeatedly
- add some words for each of the mermaid charts, I'm not sure what we're trying to express here
- can we say what problem each rule solves? I think if I knew that, I could do the rest of the updates and then request your review.

File: docs/rules/spot/parameters-not-in-body.md (hunk 8 above)
Comment (DmitryAnansky) [reply docs/rules/spot/parameters-not-in-body.md:39]: Current Spot versions going with v 0.7.x.

File: docs/rules/arazzo/parameters-unique.md (hunk 6 above)
Comment (tatomyr) [reply docs/rules/arazzo/parameters-unique.md:89]: Then checking the uniqueness with the rules still makes sense.

File: docs/rules/arazzo/parameters-unique.md [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+
+# parameters-unique
+
+Requires the `parameters` list not to include duplicate parameters.
+
+| ARAZZO | Compatibility |
+| ------ | ------------- |
+| 1.0.0  | ✅            |
+
+```mermaid
```
[the remark this answers is not in the document]
Comment (DmitryAnansky) [reply docs/rules/arazzo/parameters-unique.md:13]: Well, I was not  familiar with rule document structure, so I tried to document it similar to existing examples.
And some of them showing the diagram of validating object location inside described object.

So I tried to illustrate the nested location of the affected by the rule objects.

File: docs/rules/arazzo/workflow-dependsOn.md [hunk 10]
```diff
@@ -0,0 +1,73 @@
+---
+slug: /docs/cli/rules/arazzo/workflow-dependsOn
+---
+
+# workflow-dependsOn
```
Comment (lornajane) [inline docs/rules/arazzo/workflow-dependsOn.md:5]: Why isn't this named `*-unique` like the others? Is it because we also check it exists?

File: docs/rules/minimal.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (tatomyr) [reply docs/rules/minimal.md:12]: I'm not sure we need separate rulesets for each specification type and flavor. Redocly CLI is smart enough to detect which specification it's working with and applies only the appropriate rules. The only issue arises with Spot-specific rules which contradict (to certain extent) with the Arazzo spec. Therefore, it might be a good idea to introduce only one new spot ruleset. If you're interested in Spot-specific rules, you can use config like the following:

```yaml
extends:
- recommended
- spot
...
```

File: docs/rules/arazzo/workflow-dependsOn.md (hunk 10 above)
Comment (DmitryAnansky) [reply docs/rules/arazzo/workflow-dependsOn.md:1]: Yes, we also verify that workflows exist.

File: docs/rules/minimal.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (lornajane) [reply docs/rules/minimal.md:12]: But they're additive. I can't extend recommended for my OpenAPI and minimal for my Arazzo, I can only have everything, or manually build up the rulesets under the per-format sections.

I'm adding documentation with copy/paste samples for how to recreate the rulesets yourselves in config as a workaround - we can update the docs again if we improve the feature in future.

File: docs/rules/recommended.md (hunk 1 above)
Comment (lornajane) [reply docs/rules/recommended.md:26]: Mmmm, I added the ruleset-templates page with all the rulesets for all the formats listed. Maybe I should have split that information onto these existing minimal and recommended documentation pages though, does anyone have an opinion?

File: packages/core/src/rules/arazzo/index.ts (hunk 3 above)
Comment (lornajane) [reply packages/core/src/rules/arazzo/index.ts:25]: Got it, and updated it in the other pages!

File: docs/rules/arazzo/parameters-unique.md (hunk 6 above)
Comment (lornajane) [reply docs/rules/arazzo/parameters-unique.md]: I adjusted the example to be more "obvious" in what it illustrates.

File: docs/rules/arazzo/parameters-unique.md [hunk 11]
```diff
@@ -0,0 +1,93 @@
+---
+slug: /docs/cli/rules/arazzo/parameters-unique
```
[the remark this answers is not in the document]
Comment (lornajane) [reply docs/rules/arazzo/parameters-unique.md]: A lot of the pages have the slug defined, and using it this way does make for URLs that case-match their filenames - which is important since many new rules use mixed camel and kebab case. It does make it more confusing to preview the site locally, but this should be fixed as a separate concern.

File: docs/rules/arazzo/parameters-unique.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (lornajane) [reply docs/rules/arazzo/parameters-unique.md]: I removed the diagrams rather than fixing them

File: docs/rules/minimal.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (lornajane) [reply docs/rules/minimal.md:12]: I think this is covered by #1724 so I'm resolving this thread and we can pick up the discussion outside this PR

File: docs/rules/async/channels-kebab-case.md [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+flowchart TD
+
+root ==> channels --> channel --> address
+
+style url fill:#codaf9,stroke:#0044d4,stroke-width:5px
+```
+
+## API design principles
+
+Channel address should be `kebab-case` (lowercase with hyphens).
```
[the remark this answers is not in the document]
Comment (lornajane) [reply docs/rules/async/channels-kebab-case.md:12]: OK, I tested this feature with different version AsyncAPI descriptions and, it's just that the naming changed a lot between 2.6 and 3.0 . I'm adding something to this page to make it clearer as I'm updating it anyway.

File: docs/guides/lint-asyncapi.md [hunk 13]
```diff
@@ -53,8 +53,8 @@ some of the built-in rules. The currently-supported rules are:
...
+- `no-channel-trailing-slash`: channel names must not have trailing slashes in their address.
```
[the remark this answers is not in the document]
Comment (lornajane) [reply docs/guides/lint-asyncapi.md:57]: Updated this part and clarified the examples for the versions.

File: docs/guides/lint-asyncapi.md (hunk 4 above)
Comment (lornajane) [reply docs/guides/lint-asyncapi.md:56]: Reformatted the overall structure and these individual rules pages.

**Comment by lornajane** [comment]
Can I get some reviews on this, please? There's one open conversation about page structure but otherwise I am pretty happy with how this turned out and kudos to @DmitryAnansky for all the hard work - both on the feature and on the docs!

**Review by adamaltman — CHANGES_REQUESTED** [review]
We're going to have to discuss this offline. I'm not in agreement to move forward with this as it is.

File: docs/rules/arazzo/parameters-unique.md [hunk 14]
```diff
@@ -0,0 +1,72 @@
+# parameters-unique
+
+Requires unique values in the `parameters` lists.
+
+| Arazzo | Compatibility |
+| ------ | ------------- |
+| 1.0.0  | ✅            |
+
+## API design principles
```
[the remark this answers is not in the document]
Comment (lornajane) [reply docs/rules/arazzo/parameters-unique.md:9]: Can we pick something in plainer English (cc @HCloward for your help!) here? How about "Reasoning" or just "Principles" ?

[the remark this answers is not in the document]
Comment (HCloward) [reply docs/rules/arazzo/parameters-unique.md:9]: I think to keep consistency I like "Principles" or even "Design principles" or "Workflow design principles"

[the remark this answers is not in the document]
Comment (adamaltman) [reply docs/rules/arazzo/parameters-unique.md:9]: My perspective is the content in all of these does not fit with the word "Principles".

[the remark this answers is not in the document]
Comment (adamaltman) [reply docs/rules/arazzo/parameters-unique.md:9]: They could be rewritten though to include a principle first:

> Reduce confusion by avoiding ambiguous configuration.

File: docs/rules/async/channels-kebab-case.md [hunk 15]
```diff
@@ -0,0 +1,78 @@
...
+
+| Option   | Type   | Description                                             |
+| -------- | ------ | ------------------------------------------------------- |
+| severity | string | Possible values: `off`, `warn`, `error`. Default `off`. |
+
+An example configuration:
+
+```yaml
+rules:
+  channels-kebab-case: error
```
[the remark this answers is not in the document]
Comment (lornajane) [reply docs/rules/async/channels-kebab-case.md:24]: After discussing this with @adamaltman , I'll update this pull request to use only `rules:` and per-API configuration. We will leave the per-format approach undocumented until we reach agreement on #1723, but we won't block this pull request for that work to be done.

File: docs/rules/recommended.md (hunk 1 above)
Comment (lornajane) [reply docs/rules/recommended.md:26]: Updated to reflect only having documentation for per-API configuration and keeping the per-format information as an internal-only implementation detail for now.

File: docs/rules/arazzo/parameters-unique.md (hunk 14 above)
[the remark this answers is not in the document]
Comment (lornajane) [reply docs/rules/arazzo/parameters-unique.md]: Updated these sections

**Comment by DmitryAnansky** [comment]
Moved docs changes to https://github.com/Redocly/redocly-cli/pull/1819
